pommerman/network/ion_server/__init__.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+"""IonServer - This library allows playing of matches on network via the
+WebSockets protocol.  
+Functions:  
+init() - If you want to run the application normally  
+run(max_players, max_matches, port, timeout, mode, ui_en=True, 
+exit_handler=True) - If you want to programatically launch the server with
+predefined parameters"""
+
+import ui
+import multiprocessing
+from . import constants
+# This is for switching between languages incase there are more
+Strings = constants.LangEN
+from . import network
+from . import match
+import time
+import random
+import signal
+import pommerman
+
+MATCHSP = []
+
+
+def _exit_handler(sp_net):
+    """Description: Return the exit handler with a reference to the sp_net
+    variable."""
+
+    def exit_handler(_s, _h):
+        """Description: Handle exiting the application."""
+        ui.info(ui.yellow, "Exiting..")
+        sp_net.terminate()
+        for i in MATCHSP:
+            i.terminate()
+        exit(0)
+
+    return exit_handler
+
+
+def init():
+    """Description: Initiate the application by asking questions."""
+    ui.info(ui.yellow, Strings.sever_starting.value)
+    port = int(ui.ask_string(Strings.server_port.value))
+    max_players = int(ui.ask_string(Strings.server_maxp.value))
+    if max_players < 4:
+        ui.fatal(ui.yellow, "Input a number greater than 4")
```

Comment:
What is this supposed to be? I am confused by `max_players`. Can you have the ui.yellow response be more descriptive of what it's looking for please?

Reply:
First of all, I haven't created the UI library so I can't make it more or less descriptive. Second, `ui.yellow` is responsible for changing the color of the the outputted text.

Reply:
About `max_players`: It should be pretty obvious. It's the max concurrent players on the server at once. (This does not include dowloading replays)

Reply:
By "make it more descriptive", I meant the "Input a number greater than 4".
By me asking you about `max_players`, it means that it's not obvious to the person who wrote most of the codebase for Pommerman and additionally has been going over your code. That should make it clear that it's NOT pretty obvious. Even the fact that you have to give a parenthetical caveat should suggest this.
Given all of that, please, please, pretty please just clarify what's going on in both the code and to the user with the ui.fatal message.

Reply:
Okay